TFNetworkRecLayer.py

```diff
@@ -6002,3 +6002,73 @@ def _get_output_native_rec_op(self, cell):
       y = y[0]
 
     return y
+
+
+class ZoneoutLSTMCell(BaseRNNCell):
+  """
+  Wrapper for tf LSTM to create Zoneout LSTM Cell.
+  This code is an adapted version of Rayhane Mamas version of Tacotron-2
+  """
+
+  def __init__(self, num_units, zoneout_factor_cell=0., zoneout_factor_output=0., name=None, **kwargs):
+    """
+    Initializer with possibility to set different zoneout values for cell/hidden states.
+    :param int num_units: 
+    :param float zoneout_factor_cell: 
+    :param float zoneout_factor_output: 
+    :param string name: 
+    :param dict kwargs: 
+    """
+    zm = min(zoneout_factor_output, zoneout_factor_cell)
+    zs = max(zoneout_factor_output, zoneout_factor_cell)
+
+    if zm < 0. or zs > 1.:
+      raise ValueError('One/both provided Zoneout factors are not in [0, 1]')
+
+    self._cell = tf.nn.rnn_cell.LSTMCell(num_units, state_is_tuple=True, name=name)
+    self._zoneout_cell = zoneout_factor_cell
+    self._zoneout_outputs = zoneout_factor_output
+    from TFNetwork import TFNetwork
+    self.is_training = TFNetwork.get_current_network().train_flag
+
+  @property
+  def state_size(self):
+    """
+    :rtype: int 
+    """
+    return self._cell.state_size
+
+  @property
+  def output_size(self):
+    """
+    :rtype: int 
+    """
+    return self._cell.output_size
+
+  def __call__(self, inputs, state, scope=None):
+    """
+    Apply ZoneoutLSTM on input with given state 
+    :param tf.Tensor inputs: input tensor to the cell
+    :param tf.nn.rnn_cell.LSTMStateTuple state: previous state of the LSTM
+    :param scope: 
+    :return: tuple of output and LSTMStateTuple 
+    :rtype: tuple[tf.Tensor, tf.nn.rnn_cell.LSTMStateTuple]
```

Comment:
The syntax for `rtype` is wrong here.
(Btw, just use PyCharm and check that it works correct.)
It should be `(tf.Tensor, tf.nn.rnn_cell.LSTMStateTuple)`.